### PR TITLE
Migrate to also using unstable tests on ci

### PR DESCRIPTION
**Pull request by Veetaha**
Unstable usually adds new lints and warnings or incompatibilities to fix asap

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -82,19 +82,71 @@ jobs:
...
+    - run: cargo +${{ matrix.channel }} test
+    - run: cargo +${{ matrix.channel }} xtask pre-cache
+
+    - if: ${{ failure() }}
+      run: |
+        curl --location --request POST \
+        'https://api.github.com/repos/rust-analyzer/rust-analyzer/issues/${{ github.event.number }}/comments' \
+        --header 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
+        --header 'Content-Type: application/json' \
+        --data-raw '{ "body": "Ci on **\`${{ matrix.channel }}\`** failed!" }'
```
Comment (bjorn3) [inline .github/workflows/ci.yaml:148]:
```suggestion
        --data-raw '{ "body": "CI on **\`${{ matrix.channel }}\`** failed!" }'
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -82,19 +82,71 @@ jobs:
...
+    - run: cargo xtask pre-cache
 
     - name: Prepare cache 2
       if: matrix.os == 'windows-latest'
       run: Remove-Item ./target/debug/xtask.exe, ./target/debug/deps/xtask.exe
 
+  # Test forward compat
+  rust-unstable:
+    if: ${{ github.event_name == 'pull_request' }}
+    name: Rust ustable
```
Comment (bjorn3) [inline .github/workflows/ci.yaml:96]:
```suggestion
    name: Rust unstable
```

**Comment by Veetaha** [comment]
Hmm, GHA actions bot cannot send a comment `Resource not accessible by integration`

**Comment by Veetaha** [comment]
Hmm, `GITHUB_TOKEN` [has only read permissions for forked repos](https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#permissions-for-the-github_token)...

**Comment by Veetaha** [comment]
@matklad I guess we will need to remove the comment creation or somehow provide a token with write acesss

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Veetaha) [reply .github/workflows/ci.yaml]: ff78e2e

**Comment by matklad** [comment]
We intentionally only use stable. Minimizing CI friction is imo more important than maybe catching some extra rust bugs. Locally, I run beta for this purpose.

**Pull request closed without merging**